### ISSUE TITLE
Tags.md link error when serving or building project

### ISSUE THREAD

**Issue by Vasperous** [squidfunk/mkdocs-material#2638]
- [x] I've read the [contribution guidelines][1] and agree with them

__I've found a bug and checked that ...__

- [x] ... the problem doesn't occur with the default MkDocs template
- [x] ... the problem is not in any of my customizations (CSS, JS, template)
- [x] ... the documentation does not mention anything about my problem
- [x] ... there are no open or closed issues that are related to my problem

## Description

After adding support for the `tags` + `tags_file ` feature the `mkdocs serve` command is reporting an issue with the links from the the `tags_file` (i.e. `tags.md`).  I have seen this previously (personally) when I accidentally link to another page using a `.html` extension instead of a `.md`.

Note that the links from the `tags.md` still work but the more tags I add the more errors that I see in the terminal when serving/building.

```
[I 210506 18:57:17 watcher:110] Running task: builder (delay: None)
INFO    -  Running task: builder (delay: None)
INFO    -  Building documentation... 
WARNING -  Documentation file 'tags.md' contains a link to 'index.html' which is not found in the documentation files. 
WARNING -  Documentation file 'tags.md' contains a link to 'index.html' which is not found in the documentation files. 
WARNING -  Documentation file 'tags.md' contains a link to 'getting-started.html' which is not found in the documentation files. 
WARNING -  Documentation file 'tags.md' contains a link to 'index.html' which is not found in the documentation files. 
WARNING -  Documentation file 'tags.md' contains a link to 'index.html' which is not found in the documentation files. 
WARNING -  Documentation file 'tags.md' contains a link to 'getting-started.html' which is not found in the documentation files. 
```

### Expected behavior

Expectation is for there to be no link errors in the tags.md.

### Actual behavior

Link errors are show in the terminal when serving/building.

### Steps to reproduce the bug

1. Create a simple mkdocs.yml with `tags` and `tags_file` support (see below with the yml file I am using)
2. Create the `tags.md` page and reference in the mkdocs.yml
3. Run `mkdocs serve`

### Package versions

* Python: `Python 3.8.5`
* MkDocs: `mkdocs, version 1.1.2`
* Material: `Version: 7.1.3+insiders.2.7.1`

### Project configuration

I removed a bunch of features etc. but no combination could do away with the error.  I tried flipping `use_directory_urls` to `true` and received less errors back but errors none the less.

``` yaml
site_name: 'Test Tags'
site_description: 'This is a test documentation description'
site_author: 'Company'

# Copyright
copyright: 'Copyright &copy; 2021'

# If serving from a website you would want 'true', for offline you want false
use_directory_urls: false

nav:
  - Home:
    - Introduction: index.md
    - Getting Started: getting-started.md
    - Tags: tags.md

theme:
  name: 'material'
  palette:
    primary: 'black'
    accent: 'blue'
  font:
    text: 'Roboto'
    code: 'Roboto Mono'

markdown_extensions:
  - meta

plugins:
  - search:
      prebuild_index: false
  - localsearch
  - tags:
      tags_file: tags.md
```

### System information

* OS: Mac OS 11.3
* Browser: Chrome and Brave

  [1]: https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md

**Comment by squidfunk** [comment]
Thanks for reporting! I think this is what https://github.com/squidfunk/mkdocs-material/issues/2593#issuecomment-832753981 was referring to. I'll look into it.

**Comment by squidfunk** [comment]
This was just fixed in the latest commit to Insiders and released as part of Insiders 2.7.2. 😊

---

_Please note that tags support is still experimental, so some things might change._
